grpc/internal/errors/errors.go

```diff
@@ -0,0 +1,113 @@
+package errors
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/keratin/authn-server/app/services"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const wwwAuthneticate = `WWW-Authenticate`
+
+type FieldError struct {
```

Comment:
My memory is fuzzy on the details that lead me down this path. I presume it's because `services.fieldError` is unexported, and we need to convert gRPC's `BadRequest_FieldViolation` back to `services.FieldErrors`. I'll export `services.fieldError`, change the code, and see how it turns up.